src/components/ReservationCalendar.jsx

```diff
@@ -87,19 +101,27 @@ class ReservationCalendar extends React.Component {
             <br></br>
             {eventInfo.event.extendedProps.lecturer_name}{" "}
             {eventInfo.event.extendedProps.lecturer_surname}
-            {
-              eventInfo.event.extendedProps.status == "presence" && <><br/>Capacity: {eventInfo.event.extendedProps.capacity}{" "}</>
-            }
+            {eventInfo.event.extendedProps.status == "presence" && (
+              <>
+                <br />
+                Capacity: {eventInfo.event.extendedProps.capacity}{" "}
+              </>
+            )}
           </p>
         </div>
       );
     else return null;
   }
   /* if bookable try to book it */
   handleEventClick = ({ event }) => {
-    if (event._def.extendedProps.color == this.state.blueColor)
-      if (window.confirm("Do you want to book this lecture?"))
-        this.props.bookLecture(event._def.extendedProps.id);
+    const { color } = event._def.extendedProps;
+    const confirmText =
+      color === this.state.blueColor
+        ? "Do you want to book this lecture?"
+        : color === "#ffc107" && "Do you want to candidate this lecture?";
+    confirmText &&
+      window.confirm(confirmText) &&
+      this.props.bookLecture(event._def.extendedProps.id);
```

Comment:
It would be better not to use `color` to decide which message to display, but more meaningful properties from `event._def.extendedProps`

Reply:
I agree. I don't know what this code does and what each color mean, I just refactored it to cut duplicated code. @huanghaihang can you tell us what each color mean so that we can rewrite this?